openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -2149,6 +2149,8 @@ mod save_restore {
             pub(super) fixed_mtrrs: [u64; 11],
             #[mesh(28)]
             pub(super) variable_mtrrs: [u64; 16],
+            #[mesh(29)]
+            pub(super) message_queues: [virt::vp::SynicMessageQueues; 2],
```

Comment:
Let's not use an array for per-VTL stuff. It's pretty unclear what it means, and it's not very extensible (an array can't change sizes between saved state versions).

Either use `vtl0_message_queues`, etc., or add some `PerVtlState` struct and have `vtl0`, `vtl1`.

Reply:
Also does this stuff get used for VTL1 at all?

Reply:
All our code certainly supports using it for VTL 1. We allow passing in arbitrary VTLs at multiple different entry points that'd end up in the message queues. Hard to say if anything actually does it in practice though.

Reply:
How about a vec and a comment? That gives us extensibility without needing to take up more than one field

Reply:
If you're going to use a `Vec`, then let's add a per-VTL state struct so that we don't end up with a bunch more vecs for per-VTL stuff later (if such state ever shows up).